sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreter.scala

```diff
@@ -73,11 +74,19 @@ case class SqlScriptingInterpreter() {
           .map(new SingleStatementExec(_, Origin(), isInternal = true))
           .reverse
         new CompoundBodyExec(
-          body.collection.map(st => transformTreeIntoExecutable(st)) ++ dropVariables)
+          body.collection.map(st => transformTreeIntoExecutable(st)) ++ dropVariables, session)
       case sparkStatement: SingleStatement =>
         new SingleStatementExec(
           sparkStatement.parsedPlan,
           sparkStatement.origin,
-          isInternal = false)
+          shouldCollectResult = true)
     }
+
+  def execute(compoundBody: CompoundBody): Iterator[Array[Row]] = {
```

Comment:
can we decouple the two parts: 1) produce `DataFrame`s from the script. 2) execute the `DataFrame`s one by one w.r.t. the order.

Reply:
We decided to do it this way because each statement has to be executed as soon as we encounter it because it can throw an exception. Very soon we are introducing `handlers/conditions` to deal with this exceptions.

Reply:
We did it the way you suggested before, but as soon as we started introducing exception handling, we figured out it's hardly extensible in the future. When SQL statements throws an exception, exception handlers need to be executed immediately, and we need interpreter for that - hence, we need to do execution within the interpreter as well.

Reply:
I'm OK with a simple implementation (collect all the result rows) as a beginning, but we should build the basic framework with proper abstraction, otherwise I don't know how to review it. The idea of `handlers` makes sense.

Reply:
For now, because we don't know which one is the last statement to do only one `collect()` and use noop sink for the rest, we decided to do collect for all statements.

Either way, it is important to execute each statement as soon as we encounter it to be able to handle errors properly. [PR introducing handlers](https://github.com/apache/spark/pull/47423) is currently work in progress and will probably explain why we did things the way we did in this PR.